File: app/classes/Card.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

public enum CardType {
    Damage,
    Armor,
    Heal
}

public class Card {
    [BsonId]
    public ObjectId Id { get; set; }

    public string Name { get; set; }
    public int Cost { get; set; }

    public CardType Type { get; set; }
    public int Value { get; set; }

    public string Description { get; set; }

    public Card(string name, int cost, CardType type, int value, string description) {
        Name = name;
        Cost = cost;
        Type = type;
        Value = value;
        Description = description;
    }

    public bool CanPlay(Hero hero) {
        return hero.Mana >= Cost;
    }

    public override string ToString() {
        return $"{Name} ({Type}), Cost: {Cost}, Value: {Value}\n{Description}";
    }
}

File: app/tests/card.py
from enum import Enum
from bson import ObjectId

class CardType(Enum):
    DAMAGE = "Damage"
    ARMOR = "Armor"
    HEAL = "Heal"

class Card:
    def __init__(self, name: str, cost: int, card_type: CardType, power: int, description: str):
        if cost < 0:
            raise ValueError("Cost cannot be negative.")
        if power < 0:
            raise ValueError("Power cannot be negative.")

        self.id = ObjectId()
        self.name = name
        self.cost = cost
        self.type = card_type
        self.power = power
        self.description = description

    def can_play(self, hero):
        return hero.mana >= self.cost
    
    def __str__(self):
        return f"{self.name} ({self.type.value}), Cost: {self.cost}, Value: {self.value}\n{self.description}"